pymc/logprob/order.py

```diff
@@ -122,7 +124,94 @@ def max_logprob(op, values, base_rv, **kwargs):
     logcdf = _logcdf_helper(base_rv, value)
 
     [n] = constant_fold([base_rv.size])
-
     logprob = (n - 1) * logcdf + logprob + pt.math.log(n)
 
     return logprob
+
+
+class MeasurableMaxNeg(Max):
+    """A placeholder used to specify a log-likelihood for a min sub-graph."""
+
+
+MeasurableVariable.register(MeasurableMaxNeg)
+
+
+@node_rewriter(tracks=[Max])
+def find_measurable_min(fgraph: FunctionGraph, node: Node) -> Optional[List[TensorVariable]]:
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    if isinstance(node.op, MeasurableMaxNeg):
+        return None  # pragma: no cover
+
+    base_var = node.inputs[0]
+
+    if base_var.owner is None:
+        return None
+
+    if not rv_map_feature.request_measurable(node.inputs):
+        return None
+
+    # Min is the Max of the negation of the same distribution. Hence, op must be Elemiwise
+    if not isinstance(base_var.owner.op, Elemwise):
+        return None
+
+    # negation is -1*(rv). Hence the scalar_op must be Mul
+    if not isinstance(base_var.owner.op.scalar_op, Mul):
```

Comment:
The check needs to be a bit more exhaustive. Something like:
```suggestion
    try:
        if not (
            isinstance(base_var.owner.op.scalar_op, Mul)
            and len(base_var.owner.inputs) == 2
            and get_underlying_scalar_constant_value(base_var.owner.inputs[1]) == -1
        ):
           return None
    except NotScalarConstantError:
        return None
```